core/src/main/c/share/vec_agg.cpp

```diff
@@ -305,23 +308,25 @@ int64_t SUM_INT(int32_t *pi, int64_t count) {
         vec.load(pi);
         bVec = vec != I_MIN;
         hasData = hasData || horizontal_count(bVec) > 0;
-        result += horizontal_add_x(select(bVec, vec, 0));
+        result += horizontal_add(select(bVec, vec, 0));
```

Comment:
Can you please elaborate on the reason for this change? I don't get it.

Reply:
Good question. The only reason is code consistency: in all other places we use `horizontal_add` which is faster. But after double checking the VCL code, I no longer think it's a good idea at least for types smaller than `long`. Let me check all cases and use `horizontal_add_x`.

Reply:
Thanks for spotting this. Fixed in cb00472505ca26083db2868e858a046a02270c08